src/gcp/client.rs

```diff
@@ -517,6 +517,7 @@ impl GoogleCloudStorageClient {
             // GCS doesn't allow empty multipart uploads
             let result = self
                 .request(Method::PUT, path)
+                .header(&CONTENT_LENGTH, "0")
                 .idempotent(true)
                 .do_put()
                 .await?;
```

Comment:
```suggestion
            let result = self.put(path, PutPayload::new(), Default::default()).await?;
```

This will help to avoid any potential divergence in future, and also perhaps be a bit more clear what is going on?

Reply:
I've updated to do this, but I also had to update the implementation of `self.put` to set `Content-Length` for empty payloads.